File: 2022/1/java/A.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class A {

    private static void solve(String[] rawInput) {
        // Guard
        if (rawInput == null || rawInput.length < 1) {
            throw new Error("Raw input is NULL!");
        }

        // System.out.println("testing input...");
        // for (String inp : rawInput) {
        //     System.out.println("Line: " + inp);
        // }

        int sum = 0;
        int maxSum = 0;
        for (String inp : rawInput) {
            if (inp == null || inp.isEmpty()) {
                sum = 0;
                continue;
            }

            int nr = Integer.parseInt(inp);
            sum += nr;
            if (sum > maxSum) maxSum = sum;
        }

        System.out.println("Max sum: " + maxSum);
    }

    private static void run(String filename) {
        // Guard
        if (filename == null || filename.isEmpty()) {
            throw new Error("Filename is NULL!");
        }

        // Open file and read it.
        String content = null;
        try {
            content = Files.readString(Paths.get(filename));
        } catch (IOException ex) {
            ex.printStackTrace();
        }

        // Now split this in multiple rows, if exist.
        String[] rawInput = content.split(System.lineSeparator());

        solve(rawInput);
    }

    private static final String DEFAULT_INPUT_FILENAME = "input";

    public static void main(String[] args) {
        // Parse argument.
        String filename = null;
        filename = args != null && args.length > 0 ? args[0] : DEFAULT_INPUT_FILENAME;

        // Run the code.
        run(filename);
    }
}

File: 2022/1/py/a.py
import argparse

def solve(raw_input):
  inp = raw_input.split('\n')
  sum = 0
  max_sum = 0
  for i in inp:
    if not i or len(i) < 1:
      sum = 0
      continue

    nr = int(i)
    sum = sum + nr
    if sum > max_sum:
      max_sum = sum

  print("max_sum: " + str(max_sum))
  

def main(filename):
  # Read file and convert to ints
  input = []
  with open(filename) as file:
    input = file.read()
    solve(input)

if __name__ == '__main__':

  # Simple argparse for the input
  parser = argparse.ArgumentParser()
  parser.add_argument("filename", type=str, nargs="?", default="input",
        help="Enter the name of the file (default = input)")
  parser.add_argument("-i", "--interactive", default="false", action="store_true", dest="interactive",
        help="If flag is set, the input is interactive")
  args = parser.parse_args()

  # Run the program
  main(args.filename)